code/src/components/History.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { useSelector } from "react-redux";
+
+import { HistoryHeading, HistoryText } from "lib/Text";
+import styled from "styled-components/macro";
+
+const HistoryContainer = styled.div`
+  @media (max-width: 668px) {
+    margin: auto;
+    margin-top: 20px;
+    margin-bottom: 20px;
+  }
+`;
+
+export const History = () => {
+  const history = useSelector((store) => store.game.history);
+
+  return (
+    <HistoryContainer>
+      {history.length > 0 && <HistoryHeading>Actions:</HistoryHeading>}
+      {history
+        .slice(-9)
+        .reverse()
+        .map((item) => (
+          <HistoryText>
+            {item.type}d {item.direction}
+          </HistoryText>
+        ))}
```

Comment:
`.slice(-9)` is understandable to limit the height of the column with the action history; however, from purely a code perspective the -9 might seem a bit arbitrary.   

`.slice(NUM_HISTORY_ITEMS_VISIBLE)` could be more clear.  Also, some visually indicator to the user that they are not viewing their entire history 👍